tensorly/decomposition/constrained_parafac.py

```diff
@@ -0,0 +1,284 @@
+import numpy as np
+import warnings
+
+import tensorly as tl
+from ..random import random_cp
+from ..base import unfold
+from ..cp_tensor import (CPTensor, unfolding_dot_khatri_rao, cp_norm,
+                         validate_cp_rank)
+from ..tenalg.proximal import admm, proximal_operator
+
+# Author: Jean Kossaifi
+#         Jeremy Cohen <jeremy.cohen@irisa.fr>
+#         Caglayan Tuna <caglayantun@gmail.com>
+
+# License: BSD 3 clause
+
+
+def initialize_constrained_parafac(tensor, rank, constraints, reg_par=None, prox_par=None, init='svd', svd='numpy_svd',
+                                   random_state=None):
+    r"""Initialize factors used in `constrained_parafac`.
+
+    Parameters
+    ----------
+
+    The type of initialization is set using `init`. If `init == 'random'` then
+    initialize factor matrices with uniform distribution using `random_state`. If `init == 'svd'` then
+    initialize the `m`th factor matrix using the `rank` left singular vectors
+    of the `m`th unfolding of the input tensor. If init is a previously initialized `cp tensor`, all
+    the weights are pulled in the last factor and then the weights are set to "1" for the output tensor.
+    Lastly, factors are updated with proximal operator according to the selected constraint(s), so that they satisfy the
+    imposed constraints (does not apply to cptensor initialization).
+
+    Parameters
+    ----------
+    tensor : ndarray
+    rank : int
+    constraints : string
+    reg_par : float
+    prox_par : float
+    random_state : {None, int, np.random.RandomState}
+    init : {'svd', 'random', cptensor}, optional
+    svd : str, default is 'numpy_svd'
+        function to use to compute the SVD, acceptable values in tensorly.SVD_FUNS
+
+    Returns
+    -------
+    factors : CPTensor
+        An initial cp tensor.
+    """
+    n_modes = tl.ndim(tensor)
+    rng = tl.check_random_state(random_state)
+    if constraints is None or len(constraints) != n_modes:
```

Comment:
This will raise an error if `constraint` is not iterable (and e.g. a string), better to check whether it's a string:
```suggestion
    if constraints is None or isinstance(constraints, str):
```